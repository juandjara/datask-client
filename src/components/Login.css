.login-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  background: #f0f0f0;
}
.login-header {
  text-align: center;
  color: rgb(51,51,51);
}
.login-form {
  border: 1px solid #e4e4e4;
  padding: 1em;
  background: #f7f7f7;
}
/* fix for firefox */
.login-form input {
  box-shadow: none;
}
