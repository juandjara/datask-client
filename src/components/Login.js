import React from 'react';
import Input from 'react-toolbox/lib/input/Input'
import Checkbox from 'react-toolbox/lib/checkbox/Checkbox'
import Button from 'react-toolbox/lib/button/Button'
import './Login.css'

const Login = () => {
  return (
    <div className="login-wrapper">
      <div style={{minHeight: '400px'}} >
        <h1 className="login-header">Open Crono</h1>
        <form onSubmit={ev => ev.preventDefault()} className="login-form shadow-z1">
          <Input label="Usuario" type="text" icon="person" required />
          <Input label="ContraseÃ±a" type="text" icon="lock" required />
          <div style={{marginTop: '1em'}}>
            <Checkbox style={{opacity: 0.5}} checked={true}
                      label="Recordarme durante 24h" />
            <Button type="submit" primary raised style={{width: '100%'}}>
              Entrar
            </Button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default Login;
