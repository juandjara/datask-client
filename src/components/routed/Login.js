import React, { Component } from 'react';
import Input from 'react-toolbox/lib/input/Input'
import Checkbox from 'react-toolbox/lib/checkbox/Checkbox'
import Button from 'react-toolbox/lib/button/Button'
import { connect } from 'react-redux'
import { authenticate } from '../../reducers/auth.reducer'
import './Login.css'

class Login extends Component {
  state = {
    rememberMe: true,
    form: {
      username: '',
      password: ''
    }
  }
  onSubmit = (ev) => {
    ev.preventDefault();
    const { form, rememberMe } = this.state;
    const { location } = this.props;
    this.props.dispatch(authenticate(form, rememberMe, location.query.next))
  }
  onChange = (text, ev) => {
    const name = ev.target.name;
    this.setState(prevState => ({
      form: Object.assign(prevState.form, { [name]: text })
    }))
  }
  onCheckboxChange = (checked) => {
    this.setState({ rememberMe: checked })
  }
  render() {
    const {loading, error} = this.props;
    const {form, rememberMe} = this.state
    return (
      <div className="login-wrapper">
        <div style={{minHeight: '400px'}} >
          <h1 className="login-header">Open Crono</h1>
          <p className="color-teal" style={{textAlign: 'center'}}>
            {loading && 'Cargando ...'}
          </p>
          <p style={{color: 'tomato', textAlign: 'center'}}>{error}</p>
          <form onSubmit={this.onSubmit} className="login-form shadow-z1">
            <Input
              label="Usuario" icon="person"
              name="username" type="text" required
              value={form.username} onChange={this.onChange} />
            <Input
              minLength={4}
              label="ContraseÃ±a" type="password"
              name="password" icon="lock" required
              value={form.password} onChange={this.onChange} />
            <div style={{marginTop: '1em'}}>
              <Checkbox
                style={{opacity: 0.5}}
                onChange={this.onCheckboxChange} checked={rememberMe}
                label="Recordarme durante 24h" />
              <Button
                primary raised style={{width: '100%'}}
                type="submit" disabled={loading}>
                {loading ? 'Cargando ...' : 'Entrar'}
              </Button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => state.auth;
export default connect(mapStateToProps)(Login);
