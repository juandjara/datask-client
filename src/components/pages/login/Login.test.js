import React from 'react'
import {Login} from './Login'
import renderer from 'react-test-renderer'
import {shallow, mount} from 'enzyme'

describe('Login', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <Login
        loading={false}
        authenticate={jest.fn()}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('matches snapshot for error', () => {
    const component = renderer.create(
      <Login
        error="Wrong credentials"
        authenticate={jest.fn()}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('matches snapshot for loading', () => {
    const component = renderer.create(
      <Login
        loading={true}
        authenticate={jest.fn()}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('calls authenticate on submit', () => {
    const component = shallow(
      <Login
        location={{query: {}}}
        authenticate={jest.fn()}
      />
    )
    const {props} = component.instance()
    component.find('form').props().onSubmit({
      preventDefault: () => {}
    })
    expect(props.authenticate).toHaveBeenCalled()
  })
  it('should change state when checkbox is toggled', () => {
    const component = mount(
      <Login />
    )
    expect(component.state('rememberMe')).toEqual(true)    
    component.find('input[type="checkbox"]').simulate('click')
    expect(component.state('rememberMe')).toEqual(false)
  })
  it('should change state when text is entered on input', () => {
    const component = mount(
      <Login />
    )
    expect(component.state('form').email).toEqual('')    
    component.find('input').first().props().onChange({
      target: {name: 'email', value: 'test'}
    })
    expect(component.state('form').email).toEqual('test')
  })
})