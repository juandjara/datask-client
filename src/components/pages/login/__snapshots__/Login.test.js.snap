// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<div
  className="login-wrapper"
>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    <h1
      className="login-header"
    >
      Open Crono
    </h1>
    <p
      className="color-primary"
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <p
      style={
        Object {
          "color": "tomato",
          "textAlign": "center",
        }
      }
    />
    <form
      className="login-form shadow-z1"
      onSubmit={[Function]}
    >
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="email"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="text"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          email
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Correo electronico
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          minLength={4}
          name="password"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="password"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          lock
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Contraseña
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        style={
          Object {
            "marginTop": "1em",
          }
        }
      >
        <label
          className=""
          data-react-toolbox="checkbox"
          onMouseEnter={undefined}
          onMouseLeave={undefined}
        >
          <input
            checked={true}
            className={undefined}
            disabled={false}
            name="rememberMe"
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
          />
          <div
            className="undefined"
            data-react-toolbox="check"
            onMouseDown={[Function]}
            style={
              Object {
                "opacity": 0.5,
              }
            }
          />
          <span
            className={undefined}
            data-react-toolbox="label"
          >
            Recordarme durante 24h
          </span>
        </label>
        <button
          className=""
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          Entrar
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login matches snapshot for error 1`] = `
<div
  className="login-wrapper"
>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    <h1
      className="login-header"
    >
      Open Crono
    </h1>
    <p
      className="color-primary"
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <p
      style={
        Object {
          "color": "tomato",
          "textAlign": "center",
        }
      }
    >
      Wrong credentials
    </p>
    <form
      className="login-form shadow-z1"
      onSubmit={[Function]}
    >
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="email"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="text"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          email
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Correo electronico
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          minLength={4}
          name="password"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="password"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          lock
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Contraseña
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        style={
          Object {
            "marginTop": "1em",
          }
        }
      >
        <label
          className=""
          data-react-toolbox="checkbox"
          onMouseEnter={undefined}
          onMouseLeave={undefined}
        >
          <input
            checked={true}
            className={undefined}
            disabled={false}
            name="rememberMe"
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
          />
          <div
            className="undefined"
            data-react-toolbox="check"
            onMouseDown={[Function]}
            style={
              Object {
                "opacity": 0.5,
              }
            }
          />
          <span
            className={undefined}
            data-react-toolbox="label"
          >
            Recordarme durante 24h
          </span>
        </label>
        <button
          className=""
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          Entrar
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login matches snapshot for loading 1`] = `
<div
  className="login-wrapper"
>
  <div
    style={
      Object {
        "minHeight": "400px",
      }
    }
  >
    <h1
      className="login-header"
    >
      Open Crono
    </h1>
    <p
      className="color-primary"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Cargando ...
    </p>
    <p
      style={
        Object {
          "color": "tomato",
          "textAlign": "center",
        }
      }
    />
    <form
      className="login-form shadow-z1"
      onSubmit={[Function]}
    >
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="email"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="text"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          email
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Correo electronico
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          minLength={4}
          name="password"
          onChange={[Function]}
          onKeyPress={undefined}
          required={true}
          role="input"
          type="password"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          lock
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Contraseña
          <span
            className={undefined}
          >
             * 
          </span>
        </label>
      </div>
      <div
        style={
          Object {
            "marginTop": "1em",
          }
        }
      >
        <label
          className=""
          data-react-toolbox="checkbox"
          onMouseEnter={undefined}
          onMouseLeave={undefined}
        >
          <input
            checked={true}
            className={undefined}
            disabled={false}
            name="rememberMe"
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
          />
          <div
            className="undefined"
            data-react-toolbox="check"
            onMouseDown={[Function]}
            style={
              Object {
                "opacity": 0.5,
              }
            }
          />
          <span
            className={undefined}
            data-react-toolbox="label"
          >
            Recordarme durante 24h
          </span>
        </label>
        <button
          className=""
          data-react-toolbox="button"
          disabled={true}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="submit"
        >
          Cargando ...
        </button>
      </div>
    </form>
  </div>
</div>
`;
